lib/terrafying/generator.rb

```diff
@@ -250,6 +250,15 @@ def template(relative_path, params = {})
 
     def output_with_children
       @children.inject(@output) { |out, c| out.deep_merge(c.output_with_children) }
+      out = @output
+      if @opts_provider
+        out['resource'].keys.each do |type|
```

Comment:
`data` items might also need the provider merged into them so we'll need to do this for both `data` and `resource` keys.